OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -69,26 +71,78 @@ public override Activity Tick(Actor self)
 			if (attackStatus.HasFlag(AttackStatus.NeedsToMove))
 				return moveActivity;
 
+			if (attackStatus.HasFlag(AttackStatus.FrozenSwitch))
+			{
+				// When shooting a non-frozen target which became frozen,
+				// this makes sure that the target line will remain even after
+				// the underlying actor is destroyed
+				self.SetTargetLine(frozenSwitchActivity.Target, Color.Red);
```

Comment:
Yeah, was was just wondering about the "this makes sure that the target line will remain even after the underlying actor is destroyed" part. The red line remains, but the unit stops shooting?